### ISSUE TITLE
quesitons about yolov3-tiny-voc.cfg

### ISSUE THREAD

**Issue by samanthawyf** [AlexeyAB/darknet#1499]
Hello, I want to train yolov3-tiny on voc dataset, but I don't find the related cfg file,  can you share it? Or should I make it myself according to yolov3-tiny.cfg? Except for the classes and filter numbers, what else do I need to revise? Thanks a lot. @AlexeyAB

**Comment by AlexeyAB** [comment]
@samanthawyf Hi,

> Except for the classes and filter numbers, what else do I need to revise?

I think it is enough.
Also you can try to recalculate anchors: 

`darknet.exe detector calc_anchors data/voc.data -num_of_clusters 6 -width 416 -height 416`

**Comment by samanthawyf** [comment]
Hi, @AlexeyAB , thanks for your reply. According to your advice, I revise the classes, filter numbers. Indeed, I also revise max_batches and steps just like yolov3-voc.cfg. As for the anchors, I meet a big problem. I try your command to recalculate anchors, but find my result is different from existing cfg files. Here is my result:
```
yolov3_coco 9 anchors
18.5772,23.5201, 44.4359,70.9271, 130.7771,75.3191, 68.8305,154.8699, 98.7181,278.5557, 190.8893,181.4958, 350.6154,152.5305, 215.4821,338.5520, 379.3934,355.3969　
yolov3_coco_tiny 6 anchors
20.6856,28.5528, 69.7722,83.7019, 96.6172,180.3143, 287.5300,143.5085, 155.9451,313.5750, 361.0585,342.7979
yolov3_voc 9 anchors
40.3128,53.0350, 65.6103,123.6124, 94.6806,215.2469, 169.2410,122.0826, 134.0053,325.7523, 216.7285,228.7580, 354.5798,205.6504, 255.2334,355.9286, 378.3310,368.1178
yolov3_voc_tiny 6 anchors
48.1264,68.5757, 83.3380,185.1250, 185.1402,152.9658, 166.6413,318.9613, 336.4462,215.0689, 350.8595,367.3759
```
Data is arranged following the instruction. I don't know why occurs difference, and can the result of  yolov3_voc_tiny for 6 anchors be used?

**Comment by AlexeyAB** [comment]
> can the result of yolov3_voc_tiny for 6 anchors be used?

Yes, you can use these anchors.

**Comment by samanthawyf** [comment]
So why is the difference? Are there any mistakes I've taken？ @AlexeyAB

**Comment by AlexeyAB** [comment]
@samanthawyf Just slightly difference algorithm for calculation anchors. There is no your mistake.

**Comment by TaihuLight** [comment]
Does yolov3-tiny-voc.cfg could use the same 9 anchors as yolov3-voc?

**Comment by AlexeyAB** [comment]
@TaihuLight 
Yes, you can , but you should specify masks to distrubute all anchors across yolo-layers:

```
...

[yolo]
mask = 5,6,7,8
...

[yolo]
mask = 0,1,2,3,4
```

**Comment by TaihuLight** [comment]
@AlexeyAB Thank you.
I train yolov3-tiny-voc on voc2007+2012 with following cfg file (containing six anchors), its minimum loss is 1.8 and mAP is 58.43@IUO=0.5.
There is wrong in my cfg?
[yolov3-tiny-voc.cfg.zip](https://github.com/AlexeyAB/darknet/files/2889379/yolov3-tiny-voc.cfg.zip)

**Comment by AlexeyAB** [comment]
@TaihuLight Last cfg looks like good.

**Comment by Seiyaku** [comment]
> @AlexeyAB Thank you.
> I train yolov3-tiny-voc on voc2007+2012 with following cfg file (containing six anchors), its minimum loss is 1.8 and mAP is 58.43@IUO=0.5.
> There is wrong in my cfg?
> [yolov3-tiny-voc.cfg.zip](https://github.com/AlexeyAB/darknet/files/2889379/yolov3-tiny-voc.cfg.zip)

@AlexeyAB @TaihuLight
Hi,friend I used your cfg for training but my map is only 43% I just change  subdivision=8 .And I use yolov3-tiny.cfg is also a similar map, may I ask you to change some places? and how many iterations did you train yolov3-tiny-voc.cfgon VOC dataset? I just train 50200 iterations by default

**Comment by Seiyaku** [comment]
Because I use tiny.cfg I didn't use darknet53.conv.74 as a pre-training when training.I found no pre-training can can converge faster
